instrumentation/httpasyncclient/src/main/java/brave/httpasyncclient/TracingHttpAsyncClientBuilder.java

```diff
@@ -131,7 +134,7 @@ static void parseTargetAddress(HttpHost target, Span span) {
         new TracingAsyncRequestProducer(requestProducer, context),
         new TracingAsyncResponseConsumer<>(responseConsumer, context),
         context,
-        callback
+        new FutureCallbackWithCurrentSpan(tracer.currentSpan(), callback)
```

Comment:
the tracecontext here in currentSpan should be the same as what's above set in context.setAttribute right? as there's some cost to fetching it, I think we should remember or reuse the context value.